packages/autorest.typescript/test/rlcIntegration/lroRest.spec.ts

```diff
@@ -872,24 +868,19 @@ describe("LRO Rest Client", () => {
     });
 
     it("should handle deleteAsyncRelativeRetry400 ", async () => {
-      try {
-        const initialResponse = await client
-          .path("/lro/nonretryerror/deleteasync/retry/400")
-          .delete();
-
-        assert.equal(initialResponse.status, "202");
-        assert.isNotTrue(isUnexpected(initialResponse));
-        const poller = await getLongRunningPoller(client, initialResponse, {
-          intervalInMs: 0
-        });
-        await poller.pollUntilDone();
-        assert.fail("Scenario should throw");
-      } catch (error) {
-        assert.equal(
-          error.message,
-          "Polling was unsuccessful. Expected status to have a string value or no value but it has instead: 400. This doesn't necessarily indicate the operation has failed. Check your Azure subscription or resource status for more information."
-        );
-      }
+      const initialResponse = await client
+        .path("/lro/nonretryerror/deleteasync/retry/400")
+        .delete();
+
+      assert.equal(initialResponse.status, "202");
+      assert.isNotTrue(isUnexpected(initialResponse));
+      const poller = await getLongRunningPoller(client, initialResponse, {
+        intervalInMs: 0
+      });
+      const result = await poller.pollUntilDone();
+      // FIXME: Polling was unsuccessful. Expected status to have a string value or no value but it has instead: 400. This doesn't necessarily indicate the operation has failed. Check your Azure subscription or resource status for more information.
+      assert.equal(result.status, "400");
```

Comment:
I assume there will be no exception thrown if I enable the flag.

But here I met the ERROR.

Reply:
This is a failure related to unexpected body format where the status field is not a string. There is no way to understand this body so an error is thrown.

Reply:
So could you clarify the idea of resolveOnUnsuccess flag here? Which cases could be handled by lib? Which could not?


Previously I naively thought there will be no exception thrown when enabling this flag

Reply:
@joheredi Could you also review this pr?

Reply:
You're right that we are trying to minimize the number of thrown exceptions in RLCs and we provide the `isUnexpected` helper to inspect response objects for errors instead of throwing a `RestError`. However, there are situations where there is nothing you can do but to throw and one of those situations is failed parsing of the response. `@azure-rest/core-client` itself throws an exception in this case, see [[here]](https://github.com/Azure/azure-sdk-for-js/blob/ffdbb47c298497623af4790e1f085d19f7b36c50/sdk/core/core-client-rest/src/sendRequest.ts#L233). This failed parsing exception can't be handled by core-lro in any meaningful way because it could be the case that the operation is still running but the response wasn't a proper JSON for some reason. This is very unlikely to happen in reality though.

`resolveOnUnsuccessful` instructs the poller to not throw an error if the operation has been determined to be failed or canceled. Thrown errors originating from any other situation will still be surfaced, e.g. parse errors, abort errors, etc.

Reply:
Thanks for clarification, it's clear now! Some cases still failed with wordings see commit: https://github.com/Azure/autorest.typescript/pull/1739/commits/d4795e753fd23819a59b9dca9f6478c61a21f0ef